### ISSUE TITLE
Implement rate limiting for st2auth /tokens API endpoint (aka authenticate endpoint)

### ISSUE THREAD

**Issue by Kami** [StackStorm/st2#4114]
We should implement rate limiting on per user basis on the st2auth authenticate API endpoint.

There should be no need for single user to authenticate more than some reasonable amount of times in a short time frame (e.g. 10 in 60 seconds).

Authenticating more than some reasonable amount of times could put unnecessary stress on the upstream authentication backend server in some situations (e.g. when LDAP backend is used).

**Comment by armab** [comment]
I hope this rate-limiting would be switched off by default and can be enabled `on` only by those who really need it.
Otherwise when enforced it will bring much more confusion, rather than really help.

**Comment by Kami** [comment]
Yeah I also need to add that this could simply be a documentation for recommended configuration change - aka rate limiting performed by the HTTP server in-front of StackStorm (nginx in the default installation).

**Comment by armab** [comment]
Agree, documenting `nginx`-based rate-limiting sounds like the best performing and easy enough solution where user has some good options to control.

**Comment by nzlosh** [comment]
Reference: https://www.nginx.com/blog/rate-limiting-nginx/
